pkg/api/customization/oci/listers.go

```diff
@@ -0,0 +1,238 @@
+package oci
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/rancher/norman/httperror"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/containerengine"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/identity"
+
+	"encoding/json"
+)
+
+func processVcns(provider common.ConfigurationProvider, compartment string) ([]byte, httperror.ErrorCode, error) {
+	logrus.Debugf("[oci-handler] listing VCNs in compartment: %s", compartment)
+	virtualNetworkClient, err := core.NewVirtualNetworkClientWithConfigurationProvider(provider)
+	if err != nil {
+		logrus.Debugf("[oci-handler] error creating OCI Virtual Network client: %v", err)
+		return nil, httperror.ErrorCode{}, err
+	}
+	vcnRequest := core.ListVcnsRequest{
+		CompartmentId: &compartment,
+	}
+	vcnResponse, err := virtualNetworkClient.ListVcns(context.Background(), vcnRequest)
+	if err != nil {
+		httpErr := httperror.ErrorCode{}
+		if vcnResponse.RawResponse != nil {
+			httpErr.Status = vcnResponse.RawResponse.StatusCode
+			httpErr.Code = vcnResponse.RawResponse.Status
+		} else {
+			httpErr = httperror.ErrorCode{}
+		}
+		logrus.Debugf("[oci-handler] error listing VCNs with OCI Virtual Network client: %v", err)
+		return nil, httpErr, err
+	}
+
+	var vcnDisplayNames []string
+	for _, item := range vcnResponse.Items {
+		vcnDisplayNames = append(vcnDisplayNames, *(item.DisplayName))
+	}
+
+	data, err := json.Marshal(vcnDisplayNames)
```

Comment:
The error isn't being checked here so the return below would be 0 if there is an error.